test/register-bson.js

```diff
@@ -1,7 +1,65 @@
 'use strict';
 
-// TODO(NODE-3555): When we remove karma in favor of vm we can use ts-node to test src instead of lib here.
-const BSON = require('../lib/bson');
-const { ensureBuffer } = require('../lib/ensure_buffer');
-BSON.ensureBuffer = ensureBuffer;
+require('source-map-support').install({
+  hookRequire: true
+});
+
+const { inspect } = require('node:util');
+const vm = require('node:vm');
+const fs = require('node:fs');
+const path = require('node:path');
+const chai = require('chai');
+
+/**
+ * In the runInContext "web" testing instanceof checks fail
+ * since the error classes are declared in a different realm (?)
+ * So here we augment the chai assertion to fallback on checking the name property of the error instance
+ */
+chai.use(function (chai) {
+  const throwsAssertion = chai.Assertion.prototype.throw;
+  chai.Assertion.addMethod('throw', function (...args) {
+    try {
+      throwsAssertion.call(this, ...args);
+    } catch (assertionError) {
+      if (assertionError.actual?.name === assertionError.expected) {
```

Comment:
If the name of the actual equals the stringified expected, then we can pass the assertion. This is to make instanceof checks that fail still work in the "web" mode